clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -704,6 +717,22 @@ CIRGenFunction::buildSwitchCase(const SwitchCase &S, mlir::Type condType,
   llvm_unreachable("expect case or default stmt");
 }
 
+mlir::LogicalResult CIRGenFunction::buildCaseNoneStmt(const Stmt *S) {
+  // Create orphan region to skip over the case none stmts.
```

Comment:
Because you are creating an orphan region, this mean that anything emitted inside a `buildCaseNoneStmt` will never execute, right? The problem if a orphan region is that it won't get attached to anything, so it really adds no value (not even for unrecheable code analysis). If so, better just to split the current basic block A into two: B and C. A should jump to C and you emit the code in B.

Reply:
I understand your point, but if you go for the current approach you might as well skip this codegen entirely, because what you are emitting won't ever be attached to anything. I think it's safer to mimic the original codegen here, what is Clang currently doing for OG codegen?

Reply:
Perhaps we should create a SwitchOp with at least one default region and delete that at the end if it ends up unused?